compiler/native/native.go

```diff
@@ -146,6 +146,17 @@ func (c *client) WithMetadata(m *types.Metadata) compiler.Engine {
 	return c
 }
 
+// WithPrivateGitHub sets the private github client in the Engine.
+func (c *client) WithPrivateGitHub(url, token string) compiler.Engine {
+	if len(url) != 0 && len(token) != 0 {
+		privGithub, _ := setupPrivateGithub(url, token)
```

Comment:
thoughts on losing the error that might be returned by `setupPrivateGithub`?

Reply:
Long term there could be some concern but in general, I'm not concerned. If you traverse the code path for that function it never actually returns an error in the method bodies but implements the `error` return on the method definition. 

The setup function is defined here and is just returning the client from the registry package in the compiler:
https://github.com/go-vela/compiler/blob/e0c4cf92be384272c36e156f4720b130cd5c3b0f/compiler/native/native.go#L89

If you dig into the registry `New` func we never throw any errors but define the `error` return:
https://github.com/go-vela/compiler/blob/e0c4cf92be384272c36e156f4720b130cd5c3b0f/registry/github/github.go#L30-L57

There is a place in that function we could throw/catch an error but it's not implemented today. I don't know the reasoning behind that. I used the mental logic if we didn't care about the error before I would just ignore it for now. 

My best guess for a lot of lack of error handling is it's all just client initialization and we didn't find that aspect of it valuable.

Reply:
that's fair. i didn't traverse the code path but was curious on your thoughts. i'm good with it.

Reply:
Yeah, I did the traversal when I wrote it because I was going to do it originally but saw we didn't ever throw errors. 

I didn't check the go-github library to see if it error'd on anything useful. My best guess would be URL parse error but I'd have to check that library.

Reply:
Worst case it will error out down the line somewhere :this_is_fine: